nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/flowanalysis/StandardFlowAnalyzer.java

```diff
@@ -250,6 +260,28 @@ private void analyzeProcessGroup(
         processGroup.getProcessGroups().forEach(childProcessGroup -> analyzeProcessGroup(childProcessGroup, flowAnalysisRules, groupViolations, componentToRuleViolations));
     }
 
+    private boolean isWithinScope(FlowAnalysisRuleNode ruleNode, String groupIdentifier) {
+        final String ruleScope = ruleNode.getScope();
+
+        while (groupIdentifier != null) {
+            if (ruleScope == null || ruleScope.isBlank()) {
+                return true;
+            }
+
+            final HashSet<String> scopedProcessGroupIds = new HashSet<>(Arrays.asList(ruleScope.split("\\s*,\\s*")));
```

Comment:
As noted on the Jira issue, this approach should be refactored to avoid splitting during rule evaluation and instead storing the values as a List of strings.